scripts/lsp-generate-bindings.el

```diff
@@ -31,34 +31,43 @@
           ((&plist :$defs defs :properties) (json-read-from-string (buffer-string))))
     `(progn
        ,@(->> (append defs properties)
-              (-partition 2)
-              (-filter (-lambda ((_ (&plist :type)))
-                         (not (string= type "object"))))
-
-              (-map (-lambda ((type (&plist :enum)))
-                      (seq-map-indexed (-lambda (value index)
-                                         (list 'defconst (intern (format "lsp/%s-%s"
-                                                                         (s-dashed-words (substring (symbol-name type) 1))
-                                                                         (s-dashed-words  value)))
-                                               (1+ index)))
-                                       enum)))
-              (apply #'nconc))
+          (-partition 2)
+          (-filter (-lambda ((_ (&plist :type :enum)))
+                     (and (not (string= type "object"))
+                      enum)))
+          (-keep (-lambda ((type (&plist :enum)))
+                   (cons
+                    (list
+                     'defvar (intern (format "lsp/%s-lookup"
+                                      (s-dashed-words (substring (symbol-name type) 1))))
+                     (if (eq type :TextDocumentSyncKind)
+                         (apply 'vector (-map #'intern enum))
+                       (apply 'vector nil (-map #'intern enum))))
+                    (seq-map-indexed (-lambda (value index)
+                                       (list 'defconst (intern (format "lsp/%s-%s"
+                                                                (s-dashed-words (substring (symbol-name type) 1))
+                                                                (s-dashed-words  value)))
+                                        (if (eq type :TextDocumentSyncKind)
+                                            index
+                                          (1+ index))))
+                     enum))))
+          (apply #'nconc))
 
        ,(->> (append defs properties)
-             (-partition 2)
-             (-filter (-lambda ((_ (&plist :type)))
-                        (string= type "object")))
-             (-map (-lambda ((name type-data))
-                     (let* ((key (intern (substring (symbol-name name) 1)))
-                            (required (-map (lambda (field-name)
-                                              (intern (concat ":" field-name))) (plist-get type-data :required)))
-                            (params (->> (plist-get type-data :properties)
-                                         (-partition 2)
-                                         (-map #'cl-first)
-                                         (-filter (-not (-partial #'-contains? required))))))
-                       (list key required params))))
+         (-partition 2)
```

Comment:
The indent look weird?

Reply:
Yes, but this is what we get with the latest format settings.

Reply:
One of the `FIXME`s in the `.dir-locals.el` mentions how to work around this: we'll have to put the immediately-preceding sexp on its own line.